src/api/app/components/workflow_artifacts_per_step_component.rb

```diff
@@ -0,0 +1,90 @@
+    @artifacts_per_step = artifacts_per_step
+    @step = @artifacts_per_step.step
+    @artifacts = @artifacts_per_step.artifacts
+  end
+
+  def artifacts_sentence(context)
+    parsed_artifacts = JSON.parse(artifacts).deep_symbolize_keys
+
+    case step
+    when 'Workflow::Step::BranchPackageStep'
+      artifacts_for_branch_or_link_package(context, parsed_artifacts, 'Branched package from ')
+    when 'Workflow::Step::LinkPackageStep'
+      artifacts_for_branch_or_link_package(context, parsed_artifacts, 'Linked package from ')
+    when 'Workflow::Step::RebuildPackage'
+      artifacts_for_rebuild_package(context, parsed_artifacts)
+    when 'Workflow::Step::ConfigureRepositories'
+      artifacts_for_configure_repositories(context, parsed_artifacts)
+    end
+  rescue JSON::ParserError
+    nil
+  rescue ActionController::UrlGenerationError => e
+    concat("Could not render any artifact links: #{e}")
+  end
+
+  private
+
+  def artifacts_for_branch_or_link_package(context, parsed_artifacts, label)
+    source_path = context.package_show_path(project: parsed_artifacts[:source_project], package: parsed_artifacts[:source_package])
+    target_path = context.package_show_path(project: parsed_artifacts[:target_project], package: parsed_artifacts[:target_package])
+
+    tag.li do
+      concat(label)
+      concat(link_to("#{parsed_artifacts[:source_project]}/#{parsed_artifacts[:source_package]}", source_path))
+      concat(' to ')
+      concat(link_to("#{parsed_artifacts[:target_project]}/#{parsed_artifacts[:target_package]}", target_path))
+      concat('.')
+    end
+  end
+
+  def artifacts_for_rebuild_package(context, parsed_artifacts)
+    package_path = context.package_show_path(project: parsed_artifacts[:project], package: parsed_artifacts[:package])
+
+    tag.li do
+      concat('Rebuilt package ')
+      concat(link_to("#{parsed_artifacts[:project]}/#{parsed_artifacts[:package]}", package_path))
+      concat('.')
+    end
+  end
+
+  def artifacts_for_configure_repositories(context, parsed_artifacts)
+    project_path = context.project_show_path(project: parsed_artifacts[:project])
+    repositories_path = context.project_repositories_path(parsed_artifacts[:project])
+    tag.li do
+      concat('Configured ')
+      concat(link_to('repositories', repositories_path))
+      concat(' on project ')
+      concat(link_to("#{parsed_artifacts[:project]}", project_path))
+      concat(': ')
+      concat(list_of_repositories(parsed_artifacts[:repositories]))
+    end
+  end
+
+  def list_of_repositories(repositories)
+    tag.ul do
+      repositories.each do |repository|
+        concat(tag.li { repository_sentence(repository) })
+      end
+    end
+  end
+
+  def repository_sentence(repository)
+    concat(tag.span("#{repository[:name]}", class: 'font-italic'))
+    concat(" for architectures ")
+    concat(tag.span("#{repository[:architectures].to_sentence}", class: 'font-italic'))
+    concat(" for the following paths: ")
```

Comment:
`following` isn't needed here as it doesn't provide additional information. I suggest removing it:

```suggestion
    concat(" for the paths: ")
```

Reply:
done